lib/screens/devices/device_form_screen.dart

```diff
@@ -56,6 +62,53 @@ class _DeviceFormScreenState extends State<DeviceFormScreen> {
                 SizedBox(
                   height: 15,
                 ),
+                TextFormField(
+                  autofocus: true,
+                  focusNode: _idFocusNode,
+                  decoration: InputDecoration(
+                    hintText: 'ID',
+                    filled: true,
+                    fillColor: Colors.purple[50],
+                    border: OutlineInputBorder(borderSide: BorderSide.none),
+                  ),
+                  controller: _idController,
+                  validator: (value) {
+                    if (value == null || value.isEmpty) {
+                      return 'Give the device an id.';
+                    }
+
+                    return null;
+                  },
+                  onFieldSubmitted: (value) {
+                    _idFocusNode.requestFocus();
+                  },
+                ),
```

Comment:
For each send message, the peerID of the sending device is used to create a signature. So therefore i need to have the peerID in advance to proof if the signature is valid (in this case it is used in the _handleIntroductionReplyMessage to set the peer to active)

Reply:
Hmm, but shouldn't this be unnecessary because the owner of the private key is the only one who can decrypt messages that are encrypted with its public key. He is also the only one who can encrypt the message so that we decrypt it with their public key. We should only need to make sure that the encryption/decryption was successful, so just sending a hard-coded encrypted hello message and checking if it was received correctly after decryption should suffice, or am I missing something?

We could then send the name and the peer ID in the introduction messages and fill them in at this point. The only downside is that we would not know the peerId and peerName until we have actually communicated with the other party.

Reply:
Actually there are 2 problems if we don't set the peerID:

1. If the introduction reply messages comes back (check for success), we don't know to which peer it belongs -> I cannot connect the publicKey with the peerID. Imagine you add 2 devices in the beginning and all servers are not activated, then we have 2 open peer connections and when we then later retry to send the introduction message and a reply comes, we don't know from which one this came. 

2. (considering that we solved the matching problem): An attacker could try to manipulate the connection to always fail
  A -> B: introduction message which failed 
  C -> A: send introduction reply message (as A's public key and location is public - imagine random tries)
   => A thinks it worked and will send messages, B doesn't know A and will always decline messages.

Was this understandable? 😃

And having in mind that this is more a testing "issue" as usually you would scan the QR code (the server is active -> always success) and don't have to deal with this, i think it's okay to have the two extra fields.

Reply:
Thanks, I think I understand the problem.